starter-code/src/viking.js

```diff
@@ -1,11 +1,123 @@
 // Soldier
-function Soldier () {}
+function Soldier(health, strength) {
+  this.health = health;
+  this.strength = strength;
+}
+
+Soldier.prototype.attack = function() {
+  return this.strength;
+
+};
+
+Soldier.prototype.receiveDamage = function(damage) {
+  this.health -= damage;
+
+};
 
 // Viking
-function Viking () {}
+function Viking(name, health, strength) {
+  Soldier.call(this, health, strength);
+  this.name = name;
+}
+
+Viking.prototype = Object.create(Soldier.prototype);
+
+Viking.prototype.attack = function() {
+  return this.strength;
+};
+
+Viking.prototype.receiveDamage = function(damage) {
+  Soldier.prototype.receiveDamage.call(this, damage);
+  if (this.health > 0) {
+    return this.name + " has received " + damage + " points of damage";
+  }
+  else {
+    return this.name + " has died in act of combat";
+  }
+
+};
+
+Viking.prototype.battleCry = function() {
+  return "Odin Owns You All!";
+};
+
 
 // Saxon
-function Saxon () {}
+function Saxon(health, strength) {
+  Soldier.call (this, health, strength);
+}
+
+Saxon.prototype = Object.create(Soldier.prototype);
+
+Saxon.prototype.attack =  function(){
+  return this.strength;
+};
+
+Saxon.prototype.receiveDamage = function(damage) {
+  Soldier.prototype.receiveDamage.call(this, damage);
+  if (this.health > 0) {
+    return "A Saxon has received " + damage + " points of damage";
+  }
+
+  else {
+    return "A Saxon has died in combat";
+
+  }
+};
 
 // War
-function War () {}
+function War() {
+  this.vikingArmy = [];
+  this.saxonArmy = [];
+}
+
+War.prototype.addViking = function(viking) {
+  this.vikingArmy.push(viking);
+};
+
+War.prototype.addSaxon = function(saxon){
+  this.saxonArmy.push(saxon);
+};
+
+War.prototype.vikingAttack = function(){
+   randomSaxon = Math.floor(Math.random() * this.saxonArmy.length);
+   randomViking = Math.floor(Math.random() * this.vikingArmy.length);
+   fightResult = this.saxonArmy[randomSaxon].receiveDamage(this.vikingArmy[randomViking].attack());
```

Comment:
Aquí, te recomiendo que this.saxonArmy[randomSaxon] lo guardes en una variable, porque más adelante lo vuelves a usar. Y con this.vikingArmy[randomViking].attack(), lo mismo, lo guardas en una variable y esta operación es mucho más legible.